cEP-0017.md

```diff
@@ -0,0 +1,403 @@
+```
+
+### UI to ask permission for dependencies to be installed
+
+```python
+import sys
+import coalib.misc.Constants as consts
+
+
+def display_dependencies(bears):
+    print('The bears have the following dependencies')
+    for bear in bears:
+        for requirement in bear.REQUIREMENTS:
+            print(requirement.package, requirement.version)
+    choice = raw_input().lower()
+    if choice in consts.TRUE_STRINGS:
+        return True
+    elif choice in consts.FALSE_STRINGS:
+        return False
+    else:
+        sys.stdout.write("Please respond with 'yes' or 'no'")
+```
+
+### Splitting `DistributionRequirement`
+
+```python
+import distro
+from functools import lru_cache
+import platform
+from sarge import run, Capture
+from coala_utils.decorators import generate_eq, generate_repr
+
+
+@generate_eq("platform", "package", "version", "repo")
+@generate_repr()
+class PlatformRequirement:
+    """
+    This class helps keeping track of bear distribution requirements.
+    """
+
+    REQUIREMENTS = {}
+
+    def __init__(self, platform: str, package: str, version='', repo=''):
+        """
+        :param platform:  A string with the name of the distribution.
+                          Currently, the supported platforms are Debian,
+                          Ubuntu, CentOS, Fedora, Arch Linux, openSUSE/SUSE,
+                          Gentoo Linux, Darwin
+        :param package: A string with the name of the package to be installed.
+        :param version: Version string. Leave empty to specify latest version.
+        :param repo:    The repository from which the package is to be
+                        installed.
+        """
+        self.platform = platform
+        self.package = package
+        self.version = version
+        self.repo = repo
+
+    def __str__(self):
+        """
+        Just return package name, followed by version if given.
+        """
+        if self.version:
+            return "{}{} on {}".format(self.package,
+                                       self.version,
+                                       self.platform)
+        else:
+            return "{} on {}".format(self.package,
+                                     self.platform)
+
+    def install_package(self):
+        """
+        Runs the install command for the package given in a sub-process.
+        """
+        run(" ".join(self.install_command()), stdout=Capture(),
+            stderr=Capture())
+
+    def install_command(self):
+        """
+        Returns a string with the installation command, to be used by
+        "install_package()".
+
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    def is_installed(self):
+        """
+        Check if the requirement is satisfied.
+
+        :return: Returns True if satisfied, False if not.
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    @classmethod
```

Comment:
When you implement this, it should probably be a static method, or even a function outside of the class.